### PR TITLE
[AzureMonitorExporter] refactor logging (part5)

### PR DESCRIPTION
**Pull request by TimothyMothra**
Towards #34581

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.New.cs [hunk 1]
```diff
@@ -82,5 +82,83 @@ public void StatsbeatFailed(Exception ex)
...
+            if (IsEnabled(EventLevel.Error))
+            {
+                FailedToExtractActivityEvent(activityDisplayName, ex.FlattenException().ToInvariantString());
+            }
+        }
+
+        [Event(17, Message = "Failed to extract Activity Event due to an exception. Activity: {0}. {1}", Level = EventLevel.Error)]
+        public void FailedToExtractActivityEvent(string activityDisplayName, string exceptionMessage) => WriteEvent(17, activityDisplayName, exceptionMessage);
+
+        [Event(18, Message = "Maximum count of {0} Activity Links reached. Activity: {1}. ", Level = EventLevel.Informational)]
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.New.cs:155]: Should this be a warning or information?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.New.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ public void TransmissionFailed(bool fromStorage, int statusCode, ConnectionVars
             }
         }
 
-        [Event(8, Message = "{0} {1} {2}")]
+        [Event(8, Message = "{0} {1} {2}", Level = EventLevel.Warning)]
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.New.cs:50]: If telemetry is dropped, is that an error or warning?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs [hunk 3]
```diff
@@ -111,7 +111,7 @@ internal static void AddActivityLinksToProperties(Activity activity, ref AzMonLi
                     {
                         if (MaxlinksAllowed < activity.Links.Count())
                         {
-                            AzureMonitorExporterEventSource.Log.WriteInformational("ActivityLinksIgnored", $"Max count of {MaxlinksAllowed} has reached.");
+                            AzureMonitorExporterEventSource.Log.ActivityLinksIgnored(MaxlinksAllowed, activity.DisplayName);
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs:114]: Should we include `activity.Source.Name`?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs:114]: Do you think we need both? I thought at a minimum, collect the name of the activity.

```csharp
var activitySource = new ActivitySource("activitySourceName");
var activity = activitySource.StartActivity("activityDisplayName");
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.New.cs (hunk 1 above)
Comment (cijothomas) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.New.cs:155]: Please make the message more user friendly/actionable.
What happened? What is the impact (do we drop whole span or whole links or just the excess links)? How to mitigate (if applicable)

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.New.cs (hunk 2 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.New.cs:50]: this should be an error

**Review by rajkumar-rangaraj — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs (hunk 3 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs]: added source name

**Pull request merged**